### DISCUSSION TITLE
Question: Additional Jacobian Term in SST Axisymmetric Source Code

### DISCUSSION THREAD

**Discussion by mskim0401** [su2code/SU2#1367] [Q&A] [answered]
Hi.
I'm interested in axisymmetric problem and I'm reviewing relevant source codes.
In the SST axisymmetric source term, PR #1195 , I'm curious why additional Jacobian term commented out. 
[Here](https://github.com/su2code/SU2/pull/1195/commits/dd1b1b39e0ea33a85dc06d8e415d2e242d4ce410)

In my opinion, additional Jacobian term should be needed. 
`ResidualAxisymmetric` function may not affect the Jacobian at that time. Because after the `ResidualAxisymmetric` function, Jacobian_i seems to be re-declared.

In `SU2_CFD/src/numerics/turbulent/turb_sources.cpp`
```
   /*--- Contribution due to 2D axisymmetric formulation ---*/
  
   if (axisymmetric) ResidualAxisymmetric(alfa_blended,zeta);
   /*--- Implicit part ---*/
   Jacobian_i[0][0] = -beta_star*TurbVar_i[1]*Volume;
   Jacobian_i[0][1] = -beta_star*TurbVar_i[0]*Volume;
   Jacobian_i[1][0] = 0.0;
   Jacobian_i[1][1] = -2.0*beta_blended*TurbVar_i[1]*Volume;
  }
```
Above Jacobian is for 2-D planar. Is there no need to consider Jacobian for the axisymmetric?

Similarly, it seems that there are only inviscid contribution of Jacobian in `SU2_CFD/src/numerics/flow/flow_sources.cpp`

I'm not good at SU2 code structure, so please point me out what I missed if I'm wrong.
Thank you.

**Comment by FlorianDm** [comment, marked as answer]
Why do you believe these jacobian terms "should be needed"? 

In general a jacobian does not need to be exact and an approximation which captures the dominant terms (typically 2D convection here) my be preferable (using the default Kyrlov solver FGMRES with the ILU preconditioner the matrix does not necessarily need to be diagonally dominant, positive definite or well conditioned but it is often advantageous for convergence).

You are right that the jacobian entries are overwritten after the function call so you would need to change `=-` to `-=` there but I have tried that and found ever so slightly inferior convergence when including the turbulence axisymmetry terms (keeping all or only diagonals or only net negative contributions etc.)

**Reply by mskim0401** [reply]
I'm sorry to ask a question with my short knowledge.
It would have been a rude question. Again, I'm sorry and thank you to guide me!

**Reply by bigfooted** [reply]
please, don't apologize for learning. 

This is the best type of question: you showed that you've studied the material and you could not find the answer to your question, because the answer is based on (undocumented) experience.
